magellan-journey-sample/src/main/java/com/wealthfront/magellan/sample/Expedition.kt

```diff
@@ -1,16 +1,44 @@
 package com.wealthfront.magellan.sample
 
 import android.content.Context
+import android.util.Log
 import com.wealthfront.magellan.core.Journey
+import com.wealthfront.magellan.core.Navigable
+import com.wealthfront.magellan.navigation.NavigationPropagator.addNavigableListener
+import com.wealthfront.magellan.navigation.NavigationPropagator.removeNavigableListener
+import com.wealthfront.magellan.navigation.NavigationTraverser
+import com.wealthfront.magellan.navigation.NavigableListener
 import com.wealthfront.magellan.sample.databinding.ExpeditionBinding
 
-class Expedition : Journey<ExpeditionBinding>(ExpeditionBinding::inflate, ExpeditionBinding::container) {
+class Expedition : Journey<ExpeditionBinding>(
+  ExpeditionBinding::inflate,
+  ExpeditionBinding::container
+), NavigableListener {
```

Comment:
Nit: I think we should create an anonymous implementation of `NavigableListener` instead. Especially in a demo app, it's hard to tell what methods are from which interface when there's one class implementing them all

Reply:
On second thought, we should absolutely have this as a separate class and attach it to the lifecycle via `by lifecycle()`

Reply:
done